.changeset/dull-lamps-retire.md

```diff
@@ -0,0 +1,11 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+`Hds::DialogPrimitive`
+- Fixed error in `Description` and `Body` subcomponents, by not passing the `args` argument from the constructor to `super` 
+- Added missing arguments in `Arg` TypeScript signature object
+
+`Hds::Modal`
+- Fixed error in `Body` subcomponent, caused by not passing the `args` argument from the constructor to `super`
+- Added missing arguments in `Arg` TypeScript signature object
```

Comment:
The changelog would need to be updated, as we're not making any changes to the `DialogPrimitive` and there are no missing arguments anymore.